channeldb/migrations.go

```diff
@@ -683,3 +684,154 @@ func migrateGossipMessageStoreKeys(tx *bbolt.Tx) error {
 
 	return nil
 }
+
+// migrateOutgoingPayments moves the OutgoingPayments into a new bucket format
+// where they all reside in a top-level bucket indexed by the payment hash. In
+// this sub-bucket we store information relevant to this payment, such as the
+// payment status.
+//
+// Since the router cannot handle resumed payments that have the status
+// InFlight (we have no PaymentAttemptInfo available for pre-migration
+// payments) we delete those statuses, so only Completed payments remain in the
+// new bucket structure.
+func migrateOutgoingPayments(tx *bbolt.Tx) error {
```

Comment:
How do we feel about moving every migration in its own file? `no9_migrate_outgoing_payment.go`. And also move the legacy serialization into that file. So that if we ever decide to require a minimum upgrade version, we can easily delete all the then unneeded migrations and supporting code.

Reply:
Would prefer to do in a follow-up.

Reply:
It is not a blocker for me, but I think it is unlikely that that follow-up will happen given our other priorities.

Reply:
Sure that you don't want to do it now that the migration_09 file is there?

Reply:
I think it warrants its own PR+discussion since it is more a structural project change.